WalletWasabi/Crypto/Strobe.cs

```diff
@@ -0,0 +1,318 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace WalletWasabi.Crypto
+{
+	// https://strobe.sourceforge.io/papers/strobe-20170130.pdf
+	// Based on Merlin framework small implementation: https://doc-internal.dalek.rs/src/merlin/strobe.rs.html
+	public class Strobe128
+	{
+		private readonly static byte DDATA = 0x04; // for compatibility with cSHAKE (?)
```

Comment:
if I'm not mistaken the compatibility with cSHAKE means that even though STROBE uses the Keccak permutation function directly the same sequence of absorb/squeeze operations will result in the same hash outputs as cSHAKE, the version SHAKE that supports customization (i.e. adding a domain separator).